rfcs/2022-03-web-components.md

```diff
@@ -0,0 +1,101 @@
+
+It was discovered when working with native Web Components, there's a hydration step that needs to happen on the component that causes the CSS to sort of "detach" from the styles applied within, resulting in an ugly component. The only way to get around this was to import our CSS into each Web Component, and because our styles aren't split up by component that meant **all Design System CSS** needed to be imported into each component.
+
+It should be noted, these styles didn't include our web fonts, so it's very likely that would be another thing that would need to be imported per component.
+
+**tldr; these components were chonky.**
+
+### Preact
+
+[Preact](https://preactjs.com/) is Javascript framework that is a React alternative that uses the same API as React.
+
+For this method, Preact was installed in the Design System repository as a one-to-one replacement for React per their [migration guide](https://preactjs.com/guide/v10/switching-to-preact). Issues were encountered right out of the gate with mis-matched Typescript typing support and after researching how to resolve these, other Preact quirks were discovered.
+
+Because Preact is an entirely separate JS framework to React, it was determined that the migration and down-stream consequences would be too disruptive to consuming teams.
+
+#### Issues Encountered
+
+**Typescript typings**
+
+The CMS Design System uses Typescript and some React-specific types. These React-specific types did not translated 1:1 to Preact and it was difficult to determine what equivalent Preact typings to use for a migration.
+
+**Third-party libraries**
+The CMS Design System uses several third-party libraries for functionality and increased accessibility support. Some of these libraries are React-based. If we switched to Preact, we would have to evaluate each library and ensure that it still works as expected or find a Preact or vanilla JS alternative.
+
+**Consumption of Preact components**
+If the Design System was to migrate to Preact, this could change the way React applications would consume these components. This is not something we fully explored during this discovery, but wanted to call it out as a future risk for this implementation path. The DS team wants to minimize disruptions to consuming apps.
+
+### Web Component Frameworks (Lit & Stencil)
+
+Both [Lit](https://lit.dev/) & [Stencil](https://stenciljs.com/) were explored as Web Component-specific frameworks. Both of these libraries provide tooling to develop Web Components from scratch.
+
+It should be noted that either of these libraries would be a good choice if we wanted to build web components from the ground up, instead of trying to use existing component code. Stencil, especially, supports bindings that would allow web components to be compatible with popular JS frameworks like Vue, Angular and React. This would allow the team to maintain one code base of components (as web components) but support a wider range of frameworks.
```

Comment:
Lit also provides this compatibility and functionality for being wrapped in react, but it's very new and wraps in a direction we don't want to go..  RFC looks good! :+1: